### ISSUE TITLE
transit_model fails to open zipped file. 

### ISSUE THREAD

**Issue by crocme10** [CanalTP/transit_model#827]
When I execute the following program:
```rust
use std::path::Path;

fn main() {
    let input = Path::new("fr-sw_ntfs.zip");
    let _ = transit_model::ntfs::read(&input).unwrap();
}
```

It terminates with 

```
▶ ./target/release/test-transit-model
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ErrorMessage { msg: "comment.txt: comment_id=OBX:1 not found" }

impossible to read zipped ntfs "fr-sw_ntfs.zip"', src/main.rs:5:47
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Cargo.toml:
```toml
....
[dependencies]
transit_model = "0.42.0"
```

**Comment by crocme10** [comment]
On peut obtenir le fichier fr-sw_ntfs.zip [ici](https://navitia.opendatasoft.com/explore/dataset/fr-sw/files/7a590c56356ca7555f7f39b538426636/download/)

**Comment by datanel** [comment]
the comment id `OBX:1` in `comments.txt` is missing.
It's a data problem.

**Comment by woshilapin** [comment]
Line 1164 of the file `comment_links.txt`, there is the following line.

```csv
object_id,object_type,comment_id
StopTime:OBX:29672723-2022_HIVER-TR_B01-Samedi-01-1:0,stop_time,OBX:1
```

This means that `OBX:1` is supposed to be a `comment_id`. But if you look inside the file `comments.txt`, there is no comment with such an identifier. However, it seems to be a company identifier, as seen in line 16 of `companies.txt`.

I'm going to close this ticket since it's a data problem, not a `transit_model`. Please reopen if you can find some other information.

**Comment by woshilapin** [comment]
Or do you think `transit_model` should be more resilient and just ignore such errors and keep reading what can be read?

**Comment by crocme10** [comment]
I guess it depends on use cases and error frequencies.
In my use case, I don't care about comments and such, so, yes I would like a warning and that transit model continues, especially if comment errors are frequent.
But this is a slippery slope, and it's probably easier not to get dragged into it, and just don't do fuzzy validation.
Otherwise you would need a list of acceptable errors?

**Comment by woshilapin** [comment]
I'll reopen the ticket and we'll try to fix it. Lately, we've been in a strategy to accept a maximum of data and log what can't be read: basically trying to avoid panic on incorrect data as much as possible.

**Comment by woshilapin** [comment]
I opened https://github.com/CanalTP/transit_model/pull/828 which should fix the problem. Can you try the branch on your particular use case?

**Comment by ArnaudOggy** [comment]
Fix available in version 0.42.1 https://crates.io/crates/transit_model/0.42.1
